Hw10 hibernate

littlegreenman82 opened the pull request.

littlegreenman82 committed:
hw10 hibernate

littlegreenman82 committed:
tests

stvort commented on hw10-hibernate/src/main/java/ru/otus/hw10/hibernate/executor/DbExecutorImpl.java:
```diff
@@ -0,0 +1,122 @@
+
+            logger.info("Object loaded {}", objectData);
+        } catch (Exception e) {
+            logger.error("Error loading {}", objectData);
```
Тут было бы тоже хорошо откатить транзакцию

stvort commented on hw10-hibernate/src/main/java/ru/otus/hw10/hibernate/executor/DbExecutorImpl.java:
```diff
@@ -0,0 +1,122 @@
+
+        try {
+            sessionManager.beginSession();
+            objectData = sessionManager.getCurrentSession().getHibernateSession().get(clazz, id);
```
Ну и явно ее закоммитить тоже можно

stvort commented on hw10-hibernate/src/main/java/ru/otus/hw10/api/DbExecutor.java:
```diff
@@ -0,0 +1,16 @@
+
+import ru.otus.hw10.hibernate.exception.DbExecutorException;
+
+public interface DbExecutor<T> {
```
DbExecutor тут выступает в роли Dao. А они обычно пишутся для каждой сущности отдельно, без `<T>`. Как и сервисы. В примере так и было если посмотреть. Так же обратите внимание, что с помощью метода load, у `DbExecutor<T>` можно достать из БД любую сущность. Не только T

littlegreenman82 committed:
fixes:

1. removed DbExecutor interface and implementation
2. Dao interface was created and implemented
3. fixed lazy init for User entity

littlegreenman82 committed:
fixes:

Re-throw exception in dao with specific DaoException
